airbyte-cdk/python/airbyte_cdk/sources/streams/utils/stream_helper.py

```diff
@@ -34,7 +34,15 @@ def get_first_record_for_slice(stream: Stream, stream_slice: Optional[Mapping[st
     :raises StopIteration: if there is no first record to return (the read_records generator is empty)
     :return: StreamData containing the first record in the slice
     """
+
+    if hasattr(stream.exit_on_rate_limit, "setter"):  # Safely set the value of exit_on_rate_limit to True
+        stream.exit_on_rate_limit = True  # type: ignore[misc]
+
     # We wrap the return output of read_records() because some implementations return types that are iterable,
     # but not iterators such as lists or tuples
     records_for_slice = iter(stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice))
+
+    if stream.exit_on_rate_limit:
```

Comment:
What happens if the value was already `True` at line 38? It feels like we should set `stream.exit_on_rate_limit = previous_value`

Also, do we also need to check `hasattr(stream.exit_on_rate_limit, "setter")` here instead of just `stream.exit_on_rate_limit`?

Reply:
> What happens if the value was already True at line 38? It feels like we should set stream.exit_on_rate_limit = previous_value
I agree; this will make the function more universal.

> Also, do we also need to check hasattr(stream.exit_on_rate_limit, "setter") here instead of just stream.exit_on_rate_limit?
Yes, because the Stream class has exit_on_rate_limit in read-only mode.